build_test.go

```diff
@@ -1318,6 +1318,18 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 				})
 
 				when("builder is untrusted", func() {
+					when("building Windows containers", func() {
```

Comment:
i would've expected more unit tests, given that the degree of support this adds, but I guess the existing unit tests create Windows containers already?

Reply:
Most of the new functionality is covered by acceptance tests. Because pack uses a generic `TarWriter` interface to write layers (which is swapped out based on the target container OS) there's not a lot of extra tests to write other than checking "wiring", which is covered by acceptance.

The code for erroring out when a builder is untrusted was not explicitly covered by acceptance, so we added it here.